localstack/services/sns/filter.py

```diff
@@ -0,0 +1,429 @@
+
+            case {**kwargs}:
+                if len(kwargs) != 1:
+                    raise InvalidParameterException(
+                        f"{self.error_prefix}FilterPolicy: Only one key allowed in match expression"
+                    )
+
+                operator, value = None, None
+                for k, v in kwargs.items():
+                    operator, value = k, v
+
+                if operator in (
+                    "anything-but",
+                    "equals-ignore-case",
+                    "prefix",
+                    "suffix",
+                ):
+                    if not isinstance(value, str):
+                        raise InvalidParameterException(
+                            f"{self.error_prefix}FilterPolicy: {operator} match pattern must be a string"
+                        )
+                    return
+
+                elif operator == "exists":
+                    if not isinstance(value, bool):
+                        raise InvalidParameterException(
+                            f"{self.error_prefix}FilterPolicy: exists match pattern must be either true or false."
+                        )
+                    return
+
+                elif operator == "numeric":
+                    self._validate_numeric_condition(value)
+
+                else:
+                    raise InvalidParameterException(
+                        f"{self.error_prefix}FilterPolicy: Unrecognized match type {operator}"
+                    )
+
+            case _:
+                raise InvalidParameterException(
+                    f"{self.error_prefix}FilterPolicy: Match value must be String, number, true, false, or null"
+                )
+
+    def _validate_numeric_condition(self, value):
+        if not value:
+            raise InvalidParameterException(
+                f"{self.error_prefix}FilterPolicy: Invalid member in numeric match: ]"
+            )
+        num_values = value[::-1]
+
+        operator = num_values.pop()
+        if not isinstance(operator, str):
+            raise InvalidParameterException(
+                f"{self.error_prefix}FilterPolicy: Invalid member in numeric match: {operator}"
+            )
+        elif operator not in ("<", "<=", "=", ">", ">="):
+            raise InvalidParameterException(
+                f"{self.error_prefix}FilterPolicy: Unrecognized numeric range operator: {operator}"
+            )
+
+        value = num_values.pop() if num_values else None
+        if not isinstance(value, (int, float)):
+            exc_operator = "equals" if operator == "=" else operator
+            raise InvalidParameterException(
+                f"{self.error_prefix}FilterPolicy: Value of {exc_operator} must be numeric"
+            )
+
+        if not num_values:
+            return
+
+        if operator not in (">", ">="):
+            raise InvalidParameterException(
+                f"{self.error_prefix}FilterPolicy: Too many elements in numeric expression"
```

Comment:
Really? That is the error that aws will return if you place the `<` before `>`? That sounds silly! :rofl: